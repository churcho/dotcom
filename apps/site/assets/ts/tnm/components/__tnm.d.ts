// eslint-disable-next-line typescript/no-unused-vars
import { Route, Stop } from "../../__v3api";

export type TNMMode = "subway" | "bus" | "rail";

export interface TNMRoute extends Route {
  stops: TNMStop[];
}

export interface TNMStop extends Stop {
  directions: TNMDirection[];
}

type DirectionId = 0 | 1;

export interface TNMDirection {
  direction_id: DirectionId;
  headsigns: TNMHeadsign[];
}

export interface TNMHeadsign {
  name: string;
  times: TNMTime[];
  train_number: string | null;
}

export interface TNMTime {
  delay: number;
  scheduled_time: string[] | null;
  prediction: TNMPrediction | null;
}

export interface TNMPrediction {
  time: string[];
  status: string | null;
  track: string | null;
}

export interface Trip {
  "bikes_allowed?": true;
  direction_id: 0 | 1;
  headsign: string;
  id: string;
  name: string;
  shape_id: string;
}

export interface Schedule {
  "early_departure?": boolean;
  "flag?": boolean;
  pickup_type: number;
  route: Route;
  stop: TNMStop;
  stop_sequence: number;
  time: string;
  trip: Trip;
}

interface StopWithRoutes {
  distance: string;
  stop: TNMStop;
  routes: RouteGroup[];
}

type RouteGroupName = "commuter_rail" | "subway" | "bus" | "ferry";

interface RouteGroup {
  group_name: RouteGroupName;
  routes: TNMRoute[];
}
