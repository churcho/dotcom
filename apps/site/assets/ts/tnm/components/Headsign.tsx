import React, { ReactElement } from "react";
import { RouteType, TNMHeadsign, TNMTime, TNMPrediction } from "./__tnm";

interface Props {
  headsign: TNMHeadsign;
  routeType: RouteType;
  condensed: boolean;
}

const Headsign = (props: Props): ReactElement<any> => {
  const { headsign, routeType, condensed } = props;
  return (
    <div className={headsignClass(condensed)}>
      <div className="m-tnm-sidebar__headsign">
        {renderHeadsignName(props)}

        {routeType === 2 && renderTrainName(`Train ${headsign.train_number}`)}
      </div>
      <div className="m-tnm-sidebar__schedules">
        {headsign.times.map(time => renderTime(time, headsign.name, routeType))}
      </div>
    </div>
  );
};

const headsignClass = (condensed: boolean): string => {
  if (condensed === true) {
    return "m-tnm-sidebar__headsign-schedule m-tnm-sidebar__headsign-schedule--condensed";
  }
  return "m-tnm-sidebar__headsign-schedule";
};

const renderHeadsignName = ({
  headsign,
  routeType,
  condensed
}: Props): ReactElement<any> => {
  const modifier = !condensed && routeType == 3 ? "small" : "large";

  const headsignNameClass = `m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--${modifier}`;

  if (headsign.name.includes(" via ")) {
    const split = headsign.name.split(" via ");
    return (
      <>
        <div className={headsignNameClass}>{split[0]}</div>
        <div className="m-tnm-sidebar__via">{`via ${split[1]}`}</div>
      </>
    );
  }
  return <div className={headsignNameClass}>{headsign.name}</div>;
};

const renderTrainName = (trainName: string): ReactElement<any> => (
  <div className="m-tnm-sidebar__headsign-train">{trainName}</div>
);

const renderTime = (
  tnmTime: TNMTime,
  headsignName: string,
  routeType: RouteType
) => {
  const { prediction, scheduled_time } = tnmTime;
  const time = prediction ? prediction.time : scheduled_time;

  return (
    <div
      key={`${headsignName}-${scheduled_time.join("")}`}
      className="m-tnm-sidebar__schedule"
    >
      {routeType === 2
        ? renderTimeCommuterRail(time, prediction)
        : renderTimeDefault(time)}
    </div>
  );
};

const renderTimeDefault = (time: Array<string>): ReactElement<any> => {
  const [num, _space, mins] = time;
  return (
    <div className="m-tnm-sidebar__time">
      <div className="m-tnm-sidebar__time-number">{num}</div>
      <div className="m-tnm-sidebar__time-mins">{mins}</div>
    </div>
  );
};

const renderTimeCommuterRail = (
  time: Array<string>,
  prediction: TNMPrediction | null
): ReactElement<any> => (
  <div className="m-tnm-sidebar__time m-tnm-sidebar__time--commuter-rail">
    <div className="m-tnm-sidebar__time-number">{time.join("")}</div>
    <div className="m-tnm-sidebar__status">
      {prediction ? prediction.status : "On Time"}
    </div>
  </div>
);

export default Headsign;
