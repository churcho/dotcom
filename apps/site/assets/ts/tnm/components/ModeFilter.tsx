import React, { ReactElement } from "react";
import { clickModeAction, Dispatch } from "../state";
import { TNMMode } from "./__tnm";
import ModeIcon from "./ModeIcon";

interface Props {
  dispatch: Dispatch;
  selectedModes: TNMMode[];
}

interface ModeButtonProps {
  mode: TNMMode;
  icon: string;
  name: string;
}

interface TNMModeByV3ModeType {
  [s: number]: TNMMode;
}

export const tnmModeByV3ModeType: TNMModeByV3ModeType = {
  0: "subway",
  1: "subway",
  2: "rail",
  3: "bus"
};

export const ModeFilter = ({
  dispatch,
  selectedModes
}: Props): ReactElement<HTMLElement> => {
  const handleClickMode = (mode: TNMMode): void => {
    const updatedModes = selectedModes.includes(mode)
      ? selectedModes.filter(existingMode => !(existingMode === mode))
      : [...selectedModes, mode];

    dispatch(clickModeAction(updatedModes));
  };

  const ModeButton = ({
    mode,
    icon,
    name
  }: ModeButtonProps): ReactElement<HTMLElement> => (
    <button
      className={`btn btn-secondary btn-sm m-tnm-sidebar__filter-btn ${
        selectedModes.includes(mode) ? "active" : "inactive"
      }`}
      onClick={() => handleClickMode(mode)}
      type="button"
      aria-label={
        selectedModes.includes(mode)
          ? `remove filter by ${mode}`
          : `add filter by ${mode}`
      }
    >
      <ModeIcon type={icon} />
      {name}
    </button>
  );

  return (
    <div className="m-tnm-sidebar__filter-bar">
      <div className="m-tnm-sidebar__filter-bar-inner">
        <span className="m-tnm-sidebar__filter-header">Filter</span>
        <ModeButton mode="subway" icon="subway" name="Subway" />
        <ModeButton mode="bus" icon="bus" name="Bus" />
        <ModeButton mode="rail" icon="commuter_rail" name="Rail" />
      </div>
    </div>
  );
};
