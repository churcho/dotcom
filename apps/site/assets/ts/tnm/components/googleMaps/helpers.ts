// @ts-ignore
import stopMarker from "../../../../static/images/icon-map-stop-marker.svg";
// @ts-ignore
import stationMarker from "../../../../static/images/icon-map-station-marker.svg";
// @ts-ignore
import selectedStopMarker from "../../../../static/images/icon-map-stop-marker-hover.svg";
// @ts-ignore
import selectedStationMarker from "../../../../static/images/icon-map-station-marker-hover.svg";
// @ts-ignore
import currentLocationMarker from "../../../../static/images/icon-current-location-marker.svg";
import { iconSize } from "../../../../js/google-map/helpers";

import { MarkerData } from "../__tnm";

interface GoogleMapMarkerSVG {
  url: string;
  scaledSize: google.maps.Size;
  origin: google.maps.Point;
  anchor: google.maps.Point;
  labelOrigin: google.maps.Point;
}

export const stopIcon = (selected: boolean): string =>
  selected ? selectedStopMarker : stopMarker;
export const stationIcon = (selected: boolean): string =>
  selected ? selectedStationMarker : stationMarker;

export const buildIconFromSVG = (
  svg: string,
  size: string
): GoogleMapMarkerSVG => {
  const sizeInt = iconSize(size);

  const encoded = window.btoa(svg);

  return {
    url: `data:image/svg+xml;base64, ${encoded}`,
    scaledSize: new window.google.maps.Size(sizeInt, sizeInt),
    origin: new window.google.maps.Point(0, 0),
    anchor: new window.google.maps.Point(sizeInt / 2, sizeInt / 2),
    labelOrigin: new window.google.maps.Point(sizeInt / 2, sizeInt / 2)
  };
};

export const buildMarkerIcon = (
  data: MarkerData,
  selected: boolean
): GoogleMapMarkerSVG | undefined => {
  if (data.icon) {
    if (data.icon!.includes("station")) {
      return buildIconFromSVG(stationIcon(selected), data.size);
    }
    if (data.icon!.includes("stop")) {
      return buildIconFromSVG(stopIcon(selected), data.size);
    }
    if (data.icon!.includes("current-location")) {
      return buildIconFromSVG(currentLocationMarker, data.size);
    }
  }
  return undefined;
};
