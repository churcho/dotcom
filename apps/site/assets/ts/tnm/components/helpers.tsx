import React, { KeyboardEvent } from "react";

export const handleEnterKeyPress = (
  e: KeyboardEvent,
  onClick: Function
): void => {
  if (e.key === "Enter") {
    onClick();
  }
};

export const renderSvg = (className: string, svgText: string): JSX.Element => (
  <span
    className={className}
    // eslint-disable-next-line react/no-danger
    dangerouslySetInnerHTML={{ __html: svgText }}
  />
);
