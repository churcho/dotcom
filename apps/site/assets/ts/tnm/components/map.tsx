import React from "react";
import ReactDOM from "react-dom";
// @ts-ignore: Not typed
import GoogleMap from "../../../js/google-map-class";

interface Props {
  initialData: any;
  mapElementId: string;
}

class TransitNearMeMap extends React.Component<Props> {
  public componentDidMount(): void {
    const { mapElementId, initialData } = this.props;
    this.map = new GoogleMap(mapElementId, initialData);
    this.map.bind(this);
    if (initialData.markers.length > 0) {
      this.tightenBounds();
    }
  }

  /* eslint-disable typescript/no-explicit-any */
  public data: any;

  public map: any;
  /* eslint-enable typescript/no-explicit-any */

  public tightenBounds(): void {
    this.map.resetBounds(["current-location", "radius-west", "radius-east"]);
  }

  public render(): null {
    return null;
  }
}

const render = (): void => {
  const dataEl = document.getElementById("js-tnm-map-dynamic-data");
  if (dataEl) {
    const id = dataEl.getAttribute("data-for") as string;
    const initialData = JSON.parse(dataEl.innerHTML);
    ReactDOM.render(
      <div className="m-tnm__map">
        <div id={id} />
        <TransitNearMeMap mapElementId={id} initialData={initialData} />
      </div>,
      document.getElementById("react-root")
    );
  }
};

export default render;
