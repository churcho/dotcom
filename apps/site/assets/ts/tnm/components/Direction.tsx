import React, { ReactElement } from "react";
import Headsign from "./Headsign";
import { TNMDirection, TNMRoute } from "./__tnm";

interface Props {
  direction: TNMDirection;
  route: TNMRoute;
  stopId: string;
}

export const directionIsEmpty = (dir: TNMDirection): boolean =>
  dir.headsigns.length === 0;

export const Direction = ({
  direction,
  route
}: Props): ReactElement<HTMLElement> | null => {
  if (directionIsEmpty(direction)) {
    return null;
  }

  const condensed = direction.headsigns.length === 1;

  const hideDirectionDestination =
    condensed || route.type === 0 || route.type === 1 || route.type === 2;

  return (
    <div>
      <div className="m-tnm-sidebar__direction">
        <div className="m-tnm-sidebar__direction-name u-small-caps">
          {route.direction_names[direction.direction_id]}
        </div>
        {!hideDirectionDestination && (
          <div className="m-tnm-sidebar__direction-destination">
            {route.direction_destinations[direction.direction_id]}
          </div>
        )}
      </div>
      {direction.headsigns.map(headsign => (
        <Headsign
          key={headsign.name}
          headsign={headsign}
          routeType={route.type}
          condensed={condensed}
        />
      ))}
    </div>
  );
};

export default Direction;
