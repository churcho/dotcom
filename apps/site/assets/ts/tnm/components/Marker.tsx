import React, { ReactElement } from "react";
import MapMarker from "../../app/googleMaps/Marker";
import stopIncludesModes from "../helpers/stop-includes-modes";
import { MarkerData } from "../../app/googleMaps/__googleMaps";
import { Mode } from "../../__v3api";
import { Dispatch } from "../state";
import { StopWithRoutes } from "./__tnm";

interface Props {
  map: google.maps.Map;
  data: MarkerData;
  isSelected: boolean;
  dispatch: Dispatch;
  stopData: StopWithRoutes | undefined;
  shouldFilterMarkers: boolean;
  selectedModes: Mode[];
}

export const isVisible = (
  stopData: StopWithRoutes | undefined,
  shouldFilterMarkers: boolean,
  selectedModes: Mode[]
): boolean =>
  shouldFilterMarkers && stopData
    ? stopIncludesModes(stopData, selectedModes)
    : true;

const Marker = ({
  map,
  data,
  isSelected,
  dispatch,
  stopData,
  shouldFilterMarkers,
  selectedModes
}: Props): ReactElement<HTMLElement> => (
  <MapMarker
    map={map}
    data={{
      ...data,
      "visible?": isVisible(stopData, shouldFilterMarkers, selectedModes)
    }}
    dispatch={dispatch}
    isSelected={isSelected}
  />
);

export default Marker;
