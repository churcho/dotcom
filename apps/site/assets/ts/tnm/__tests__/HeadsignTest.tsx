import React from "react";
import renderer from "react-test-renderer";
import Headsign from "../components/Headsign";
import { createReactRoot } from "./helpers/testUtils";
import { TNMHeadsign } from "../components/__tnm";

/* eslint-disable typescript/camelcase */

it("it renders 2 predictions", () => {
  const headsign: TNMHeadsign = {
    name: "Watertown",
    times: [
      {
        delay: 0,
        scheduled_time: ["4:30", " ", "PM"],
        prediction: {
          time: ["14", " ", "min"],
          status: null,
          track: null
        }
      },
      {
        delay: 0,
        scheduled_time: ["5:00", " ", "PM"],
        prediction: {
          time: ["44", " ", "min"],
          status: null,
          track: null
        }
      }
    ],
    train_number: null
  };

  createReactRoot();
  const tree = renderer
    .create(<Headsign headsign={headsign} condensed={false} routeType={1} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("it renders scheduled time when prediction is null", () => {
  const headsign: TNMHeadsign = {
    name: "Watertown",
    times: [
      {
        delay: 0,
        scheduled_time: ["4:30", " ", "PM"],
        prediction: null
      }
    ],
    train_number: null
  };

  createReactRoot();
  const tree = renderer
    .create(<Headsign headsign={headsign} condensed={false} routeType={1} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("it splits the headsign name when it contains 'via' ", () => {
  const headsign: TNMHeadsign = {
    name: "Watertown via Harvard Square",
    times: [
      {
        delay: 0,
        scheduled_time: ["4:30", " ", "PM"],
        prediction: {
          time: ["14", " ", "min"],
          status: null,
          track: null
        }
      },
      {
        delay: 0,
        scheduled_time: ["5:00", " ", "PM"],
        prediction: {
          time: ["44", " ", "min"],
          status: null,
          track: null
        }
      }
    ],
    train_number: null
  };

  createReactRoot();
  const tree = renderer
    .create(<Headsign headsign={headsign} condensed={false} routeType={1} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("it renders a status and train name for Commuter Rail", () => {
  const headsign: TNMHeadsign = {
    name: "Framingham",
    times: [
      {
        delay: 0,
        scheduled_time: ["4:30", " ", "PM"],
        prediction: {
          time: ["4:30", " ", "PM"],
          status: "On time",
          track: null
        }
      }
    ],
    train_number: "591"
  };

  createReactRoot();
  const tree = renderer
    .create(<Headsign headsign={headsign} condensed={false} routeType={2} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("it renders a status and train name for Commuter Rail with track number if available", () => {
  const headsign: TNMHeadsign = {
    name: "Framingham",
    times: [
      {
        delay: 0,
        scheduled_time: ["4:30", " ", "PM"],
        prediction: {
          time: ["4:30", " ", "PM"],
          status: "Now boarding",
          track: "1"
        }
      }
    ],
    train_number: "591"
  };

  createReactRoot();
  const tree = renderer
    .create(<Headsign headsign={headsign} condensed={false} routeType={2} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("it renders uncondensed bus headsign name as --small", () => {
  const headsign: TNMHeadsign = {
    name: "Watertown via Copley (Express)",
    times: [
      {
        delay: 0,
        scheduled_time: ["7:00", " ", "PM"],
        prediction: {
          time: ["14", " ", "min"],
          status: null,
          track: null
        }
      },
      {
        delay: 0,
        scheduled_time: ["7:10", " ", "PM"],
        prediction: {
          time: ["44", " ", "min"],
          status: null,
          track: null
        }
      }
    ],
    train_number: null
  };

  createReactRoot();
  const tree = renderer
    .create(<Headsign headsign={headsign} condensed={false} routeType={3} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("it displays delayed status for CR", () => {
  const headsign: TNMHeadsign = {
    name: "Delayed Train",
    train_number: "404",
    times: [
      {
        delay: 3,
        scheduled_time: ["7:00", " ", "PM"],
        prediction: {
          time: ["7:03", " ", "PM"],
          status: null,
          track: null
        }
      }
    ]
  };
  createReactRoot();
  const tree = renderer
    .create(<Headsign headsign={headsign} condensed={false} routeType={2} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
