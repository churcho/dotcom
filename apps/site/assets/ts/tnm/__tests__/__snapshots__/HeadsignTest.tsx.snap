// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 2 predictions 1`] = `
<div
  className="m-tnm-sidebar__headsign-schedule"
>
  <div
    className="m-tnm-sidebar__headsign"
  >
    <div
      className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--large"
    >
      Watertown
    </div>
    <div
      className="m-tnm-sidebar__via"
    >
      via Copley (Express)
    </div>
  </div>
  <div
    className="m-tnm-sidebar__schedules"
  >
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time"
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          14
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          min
        </div>
      </div>
    </div>
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time"
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          44
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          min
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders a status and train name for Commuter Rail 1`] = `
<div
  className="m-tnm-sidebar__headsign-schedule"
>
  <div
    className="m-tnm-sidebar__headsign"
  >
    <div
      className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--large"
    >
      Framingham
    </div>
    <div
      className="m-tnm-sidebar__headsign-train"
    >
      Train 591
    </div>
  </div>
  <div
    className="m-tnm-sidebar__schedules"
  >
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time m-tnm-sidebar__time--commuter-rail"
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          4:30 PM
        </div>
        <div
          className="m-tnm-sidebar__status"
        >
          On Time
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders a status and train name for Commuter Rail with track number if available 1`] = `
<div
  className="m-tnm-sidebar__headsign-schedule"
>
  <div
    className="m-tnm-sidebar__headsign"
  >
    <div
      className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--large"
    >
      Framingham
    </div>
    <div
      className="m-tnm-sidebar__headsign-train"
    >
      Train 591
    </div>
  </div>
  <div
    className="m-tnm-sidebar__schedules"
  >
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time m-tnm-sidebar__time--commuter-rail"
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          4:30 PM
        </div>
        <div
          className="m-tnm-sidebar__status"
        >
          On time track 1
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders prediction time string 1`] = `
<div
  className="m-tnm-sidebar__headsign-schedule"
>
  <div
    className="m-tnm-sidebar__headsign"
  >
    <div
      className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--large"
    >
      Watertown
    </div>
    <div
      className="m-tnm-sidebar__via"
    >
      via Copley (Express)
    </div>
  </div>
  <div
    className="m-tnm-sidebar__schedules"
  >
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time"
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          14
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          min
        </div>
      </div>
    </div>
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time"
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          10:35
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          PM
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders scheduled time when prediction is null 1`] = `
<div
  className="m-tnm-sidebar__headsign-schedule"
>
  <div
    className="m-tnm-sidebar__headsign"
  >
    <div
      className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--large"
    >
      Watertown
    </div>
    <div
      className="m-tnm-sidebar__via"
    >
      via Copley (Express)
    </div>
  </div>
  <div
    className="m-tnm-sidebar__schedules"
  >
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time"
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          14
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          min
        </div>
      </div>
    </div>
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time"
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          12:08
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          PM
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders uncondensed bus headsign name as --small 1`] = `
<div
  className="m-tnm-sidebar__headsign-schedule"
>
  <div
    className="m-tnm-sidebar__headsign"
  >
    <div
      className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--small"
    >
      Watertown
    </div>
    <div
      className="m-tnm-sidebar__via"
    >
      via Copley (Express)
    </div>
  </div>
  <div
    className="m-tnm-sidebar__schedules"
  >
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time"
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          14
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          min
        </div>
      </div>
    </div>
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time"
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          44
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          min
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it splits the headsign name when it contains 'via'  1`] = `
<div
  className="m-tnm-sidebar__headsign-schedule"
>
  <div
    className="m-tnm-sidebar__headsign"
  >
    <div
      className="m-tnm-sidebar__headsign-name m-tnm-sidebar__headsign-name--large"
    >
      Watertown
    </div>
    <div
      className="m-tnm-sidebar__via"
    >
      via Harvard Square
    </div>
  </div>
  <div
    className="m-tnm-sidebar__schedules"
  >
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time"
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          14
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          min
        </div>
      </div>
    </div>
    <div
      className="m-tnm-sidebar__schedule"
    >
      <div
        className="m-tnm-sidebar__time"
      >
        <div
          className="m-tnm-sidebar__time-number"
        >
          44
        </div>
        <div
          className="m-tnm-sidebar__time-mins"
        >
          min
        </div>
      </div>
    </div>
  </div>
</div>
`;
